platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/consensus/NonAncientEventWindow.java

```diff
@@ -76,20 +96,15 @@ public NonAncientEventWindow(
      */
     @NonNull
     public static NonAncientEventWindow getGenesisNonAncientEventWindow(@NonNull final AncientMode ancientMode) {
-        return new NonAncientEventWindow(
-                ConsensusConstants.ROUND_FIRST,
-                ConsensusConstants.ROUND_FIRST,
-                EventConstants.FIRST_GENERATION,
-                ancientMode);
+        final long firstIndicator = ancientMode == GENERATION_THRESHOLD ? FIRST_GENERATION : ROUND_FIRST;
+        return new NonAncientEventWindow(ROUND_FIRST, firstIndicator, firstIndicator, ancientMode);
     }
 
     /**
      * @return true if this is a genesis non-ancient event window, false otherwise.
      */
     public boolean isGenesis() {
-        return this.latestConsensusRound == ConsensusConstants.ROUND_FIRST
-                && this.minRoundNonAncient == ConsensusConstants.ROUND_FIRST
-                && this.minGenNonAncient == EventConstants.FIRST_GENERATION;
+        return latestConsensusRound == ROUND_FIRST;
```

Comment:
I just realized there may have been a problem with the previous definition of the genesis `NonAncientEventWindow`.  The `latestConsensusRound` is the last round that has come to consensus.   At genesis no round has come to consensus.  Perhaps the genesis settings for the NonAncientEventWindow should have `latestConsensusRound` := `ROUND_NEGATIVE_INFINITY.

Reply:
This mistake didn't break anything because it was only used for determining nonAncient and no event can be given a birthRound less than ROUND_FIRST.

Reply:
Makes sense. Done.